### PR TITLE
nontrivial changes to not reload or redownload the whitelist when we don't need to, by:

### PR DESCRIPTION
**Pull request by sapphonie**
setting mp_tournament_whitelist to the downloaded whitelist filename
setting mp_tournament_whitelist to developmentonly when tftrue_whitelist_id is not default, so that only we can change it
not reloading the whitelist if we don't actually need to if we're not using tftrue_whitelist_id
etc

This is a nontrivial change and should be thoroughly reviewed and tested b4 pushing to master

### CODE REVIEW

File: SDK.h [hunk 1]
```diff
@@ -32,6 +32,13 @@
 #include "inetchannel.h"
 #include "replay/ienginereplay.h"
 
+// we do this so we can change cvar flags easier
+#define protected public
```
Comment (AnAkkk) [inline SDK.h]: Should use the EditableConVar class in editablecommands.h

Comment (sapphonie) [reply SDK.h]: fixed in e02e967

**Comment by sapphonie** [comment]
I've gone ahead and tested this with Arie @ serveme and on my own DM servers, and it seems to work beautifully. Ready to merge whenever.

**Comment by sapphonie** [comment]
Went ahead and also added the previous connection upgrade to http 2.0, and let whitelist tf control its own caching, as http1.1+nocache uses more data for pretty much no reason.

File: items.cpp [hunk 2]
```diff
@@ -180,51 +184,108 @@ char *CItems::GetAttributeValue(KeyValues *pKItem, const char *szAttribute)
...
+		g_Items.item_whitelist->Clear();
+
 		SOCKET sock = INVALID_SOCKET;
-		if(ConnectToHost("whitelist.tf", sock))
+		if (ConnectToHost("whitelist.tf", sock))
 		{
 			int iWhiteListID = 0;
 			char szConfigURL[50];
-			char szConfigPath[50];
+			//char szConfigPath[50];
```
Comment (AnAkkk) [inline items.cpp]: duplicate commented line

**Pull request merged**